test/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoaderTest.java

```diff
@@ -3500,6 +3501,118 @@ public void testScoreboardHTMLLocations() throws Exception {
+        
+        Account[] teams = getTeamAccounts(contest);
+        assertEquals("Team count", 22, teams.length);
+
+        String displayString = contest.getContestInformation().getTeamDisplayOnScoreboard();
+        assertEquals("Team getTeamDisplayOnScoreboard", "{:teamname}", displayString);
+        
+        /**
+         * 
+     * Client/Team number - {:clientnumber} = 514
+     * Country Code - {:countrycode} = CAN
+     * CMS/External ID - {:externalid} = 309407
+     * Group Id Number - {:groupid} = 12545
+     * Group name - {:groupname} = Canada - University of British Columbia D1
+     * Long School name - {:longschoolname} = UBC! (U British Columbia)
+     * Short Shool name - {:shortschoolname} = U British Columbia
+     * Team site number - {:sitenumber} = 1
+     * Team login name - {:teamloginname} = team514
+     * Team name - {:teamname} = UBC!
+         */
+
+        Account account20 = getSortedTeamAccounts(contest)[20];
+
+        String teamDisplayString = ScoreboardVariableReplacer.substituteDisplayNameVariables(displayString, contest, account20);
+        assertEquals("Expected display ", "team21", teamDisplayString);
+        
+        yamlTestTeamDisplayOnBoard(dirname, teamDisplayString, "team21");
+    }
+    
+    public void testOne() throws Exception {
+        
+        String dataDirName = getDataDirectory(getName());
+//        Utilities.insureDir(dataDirName);
+//        assertDirectoryExists(dataDirName);
+        
+        String teamDisplayString = "{:clientnumber} {:countrycode} {:groupid} {:groupname} {:externalid}";
+        String expeccted = "21 XXX {:groupid} {:groupname} 1021";
+        yamlTestTeamDisplayOnBoard(dataDirName, teamDisplayString, expeccted);
+    }
+    
+    void yamlTestTeamDisplayOnBoard (String dirname, String teamDisplayString, String expectedString) {
+        
+        String[] yamlLines = { //
+                "",
+                "team-display-on-scoreboard-string : '"+teamDisplayString+"'", //
+                "",
+                "accounts:", //
+                "  - account: TEAM", //
+                "    site: 1", //
+                "    count: 22", //
+        };
+
+
+        IInternalContest contest = loader.fromYaml(null, yamlLines, dirname);
+        
+        Account[] teams = getTeamAccounts(contest);
+        assertEquals("Team count", 22, teams.length);
+
+        String ciDisplayString = contest.getContestInformation().getTeamDisplayOnScoreboard();
+        assertEquals("Team getTeamDisplayOnScoreboard", teamDisplayString, ciDisplayString);
+
+        Account account20 = getSortedTeamAccounts(contest)[20];
+
+        String actual = ScoreboardVariableReplacer.substituteDisplayNameVariables(ciDisplayString, contest, account20);
+        assertEquals("Expected display string for "+ciDisplayString, expectedString, actual);
+
+    }
+    
+    public void testname() throws Exception {
```

Comment:
This method is testing a lot more than just "name" -- it appears to be testing ALL substitution variables.  Shouldn't it be named something like `testAllSubstitutions`?

Reply:
will update name